### ISSUE TITLE
Remove ______ only removes 1st instance

### ISSUE THREAD

**Issue by tehhowch** [endless-sky/endless-sky#2742]
It can happen that a system's fleets definition obtains more than one instance of a particular fleet, either with the same period or different periods, e.g.:
```
system "Testbed"
    fleet "Small Southern Pirates" 2000
    fleet "Small Southern Pirates" 800
    fleet "Southern Pirate Miners" 2000
    fleet "Large Security Force" 500
```
If one wants to use the `remove fleet` syntax to edit this system, in order to remove fully remove the "Small Southern Pirates" fleet, an event will need to specify `remove fleet "Small Southern Pirates"` twice. For players with multiple mods installed, the only way a mod creator can guarantee that a specific fleet is removed is to perform a number of removals in any one event (or play 'not nice' and just redefine the system's fleets entirely).

There are a couple of ways to address this
1) forbid multiple definitions by merging successively-defined fleets (e.g. 2000 + 800 -> 571)
2) use a more complex iterative removal solution.
3) keep the current behavior

Some notes:
 - Merging instances of the same fleet would allow for partial reductions in that fleet's arrival rate
 - Merging instances prevents "flavor" behavior such as defining one of the fleet's entries with a low period (for a constant presence) and another with a rather high period (a chance opportunity).
 - Partial reductions and merging behavior simplify event definitions considerably
 - Complex iterative removals may make the code harder to read
 - Mod creators should probably know what exactly is going on w.r.t. fleets in systems at each point of their story, and thus should know how many times they need to call `remove ____` to fully remove a fleet from a system. However, they won't know if there are extras that their mod did not add.

**Comment by tehhowch** [comment]
My vote is either to implement auto-merging and partial reductions, or switch to a more complex iterative removal (an outer while loop that repeats the existing range-based for loops)

**Comment by jafdy** [comment]
The fleet arrival rates should be a Poisson distribution, one over the period can be added up and the result should be indistinguishable from leaving them separate.

**Comment by endless-sky** [comment]
To me, the existing behavior seems fine. Saying that if you add an item twice you also have to remove it twice makes at least as much sense as saying that "remove" automatically clears any number of copies of an item.

**Comment by tehhowch** [comment]
My preference is for an auto-merge to maintain the average period (e.g. 2000 + 800 -> 571) so that a single remove will do the removal, and an additional system fleet keyword, `"set period"`, which simply overwrites the existing fleet period, or adds the fleet to the system with the desired period.

**Comment by tehhowch** [comment]
This is resolved by #2956

**Comment by tehhowch** [comment]
closing based on https://github.com/endless-sky/endless-sky/pull/2956#issuecomment-336552426
